flow/hooks/track_operations.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) 2023 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Built in execution hook for basic tracking."""
+import json
+
+from .util import collect_metadata
+
+try:
+    import git
+except ImportError:
+    GIT = False
+else:
+    from .git_util import collect_metadata_with_git
+
+    GIT = True
+
+
+FN_LOGFILE = ".operations_log.txt"
+
+
+class TrackOperations:
+    """:class:`~.TrackOperations` tracks information about the execution of operations to a logfile.
+
+    This hooks can provides information on the start, successful completion, and/or
+    erroring of one or more operations in a `flow.FlowProject` instance. The logs are stored in a
+    file given by the parameter ``fn_logfile``. This file will be appended to if it already exists.
+    The default formating for the log provides the [time, job id, log level, and log message].
+    .. note::
```

Comment:
I think this requires a blank line to render correctly, but I can't tell because this isn't in the API docs yet.
```suggestion

    .. note::
```